File: programming_design_02/hw4/rank/MakeCase/output.cpp
#include<bits/stdc++.h>
#define int long long

using namespace std;




struct student{
  int grade[5];
  char name[20];
}typedef student;

bool compare(student a, student b){
  int sum1 = 0, sum2 = 0;
  for(int i = 0; i < 5; i++){
    sum1 += a.grade[i];
    sum2 += b.grade[i];
  }
  if(sum1 != sum2) return sum1 > sum2;
  for(int i = 0; i < 5; i++){
    if(a.grade[i] != b.grade[i]) return a.grade[i] > b.grade[i];
  }
  return strcmp(a.name, b.name) < 0;
}

void swap(student *a, student *b){
  student temp = *a;
  *a = *b;
  *b = temp;
}

void sort(student arr[], int n){

  for(int i = 0; i < n; i++){
    for(int j = 0; j < n - i - 1; j++){
      if(!compare(arr[j], arr[j + 1])){
        swap(&arr[j], &arr[j + 1]);
      }
    }
  }
}


  

  
void output(string caseName){
  freopen(caseName.c_str(), "r", stdin);
  string outputName = caseName;
  outputName.pop_back();
  outputName.pop_back();
  outputName += "ans";
  freopen(outputName.c_str(), "w", stdout);
  int n;
  scanf("%lld", &n);
  student num[n + 3];

  for(int i = 0; i < n; i++){
    scanf("%s%lld%lld%lld%lld%lld", num[i].name, &num[i].grade[0], &num[i].grade[1], &num[i].grade[2], &num[i].grade[3], &num[i].grade[4]);
  }

  sort(num, n);

  for(int i = 0; i < n; i++){
    printf("%s\n", num[i].name);
  }
}

signed main(){
  cin.tie(nullptr);
  string path = "../data/";
  // sample testcase
  output(path + "sample/" + "1.in");
  // secret testcase
  int testCaseName = 10;
  for(int i = 2; i <= testCaseName; i++){
    string filename = path + "secret/" + to_string(i) + ".in";
    output(filename);
  }
  return 0;
}
 

File: programming_design_01/2024/mid/pow/MakeCase/output.py
import os

def output(case_name):
    # 使用'r'模式讀取輸入檔案
    with open(case_name, 'r') as input_file:
        # 讀取檔案內容（如果需要的話）
        content = input_file.read()

    # 移除檔案名的最後三個字符（".in"）並添加".ans"後綴
    output_name = case_name[:-3] + ".ans"
    a, b = map(int, content.split())
    # 使用'w'模式寫入輸出檔案
    with open(output_name, 'w') as output_file:
        # 寫入到輸出檔案（這裡假設直接寫入讀取的內容，可根據需要修改）
        output_file.write(str(a * b))

def main():
    path = "../data/"
    # 處理樣本測試案例
    output(os.path.join(path, "sample/", "1.in"))
    output(os.path.join(path, "sample/", "2.in"))
    # 處理秘密測試案例
    test_case_name = 11
    for i in range(3, test_case_name + 1):
        filename = os.path.join(path, "secret/", f"{i}.in")
        output(filename)

if __name__ == "__main__":
    main()
